# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1537208127 -7200
#      Mon Sep 17 20:15:27 2018 +0200
# Node ID d3f5d021255d3e717ddf084449565758f5710c0e
# Parent  75ed70a1c1089ed369d4beafa58707540e1f4016
Allow building with Qt4 again (bug #49053)

* Figure.cc: don't include QWindow.h when we know it is not present

diff --git a/libgui/graphics/Figure.cc b/libgui/graphics/Figure.cc
--- a/libgui/graphics/Figure.cc
+++ b/libgui/graphics/Figure.cc
@@ -36,18 +36,20 @@ along with Octave; see the file COPYING.
 #include <QImage>
 #include <QMainWindow>
 #include <QMenu>
 #include <QMenuBar>
 #include <QMessageBox>
 #include <QtDebug>
 #include <QTimer>
 #include <QToolBar>
-#include <QWindow>
-#include <QScreen>
+#if defined (HAVE_QSCREEN_DEVICEPIXELRATIO)
+#  include <QWindow>
+#  include <QScreen>
+#endif
 
 #include "Canvas.h"
 #include "Container.h"
 #include "Figure.h"
 #include "FigureWindow.h"
 #include "MouseModeActionGroup.h"
 #include "QtHandlesUtils.h"
 
